pkg/pool-manager/pool.go

```diff
@@ -176,3 +192,42 @@ func getNextMac(currentMac net.HardwareAddr) net.HardwareAddr {
 
 	return currentMac
 }
+
+// Checks if the namespace of an instance is opted in for kubemacpool
+func (p *PoolManager) isInstanceOptedIn(namespaceName, labelSelector, labelValue string) (bool, error) {
+	log.V(3).Info("checking for label in namespace", "webhook labelSelector", labelSelector, "webhook labelValue", labelValue)
+	isNamespaceOptedIn := false
+
+	ns, err := p.kubeClient.CoreV1().Namespaces().Get(namespaceName, metav1.GetOptions{})
+	if err != nil {
+		return isNamespaceOptedIn, err
+	}
+
+	log.V(3).Info("namespaceName Labels", "namespaceName", namespaceName, "Labels", ns.GetLabels())
+	if value, ok := ns.GetLabels()[labelSelector]; ok && labelValue == value {
+		isNamespaceOptedIn = true
+	}
+
+	return isNamespaceOptedIn, err
+}
+
+func getOptInLabelValueFromMutatingWebhookConfiguration(kubeClient kubernetes.Interface, mutatingWebhookConfigName, webhookName, optInLabel string) (string, error) {
+	if mutatingWebhookConfigName == "" {
+		return "", nil
+	}
+
+	mutatingWebhookConfiguration, err := kubeClient.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Get(mutatingWebhookConfigName, metav1.GetOptions{})
+	if err != nil {
+		return "", errors.Wrap(err, fmt.Sprintf("Failed to get mutatingWebhookConfig %s", mutatingWebhookConfigName))
+	}
+
+	for _, webhook := range mutatingWebhookConfiguration.Webhooks {
+		if webhook.Name == webhookName {
+			if labelValue, ok := webhook.NamespaceSelector.MatchLabels[optInLabel]; ok && labelValue != "" {
+				return labelValue, nil
+			}
+		}
+	}
+
+	return "", errors.Wrap(err, fmt.Sprintf("label %s not found in mutatingWebhookConfig %s", optInLabel, mutatingWebhookConfigName))
```

Comment:
Do this like `errors.Wrapf(err, "label %s not found in mutatingWebhookConfig %s", optInLabel, mutatingWebhookConfigName)`